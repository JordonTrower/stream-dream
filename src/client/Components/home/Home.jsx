import React, { Component } from 'react';
// import axios from 'axios';
import styled from 'styled-components';
import Carousel from '../carousel/Carousel';

const HomeOneDiv = styled.div`
	background-color: #191B21;
	width: 100%;
	height: 100%;
	> ::-webkit-scrollbar {
		display: none;
	}
`;

const HomeLiveContainer = styled.div`
	width: 90%;
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
`;

export default class Home extends Component {
	// constructor() {
	// super();
	// this.state = {
	// 	carouselItems: []
	// 	// liveItems: [],
	// }
	// }

	componentDidMount() {
		// axios.get(`${process.env.REACT_APP_API_LOCATION}/the_carousel_things`).then(res => {
		// 	this.setState({
		// 		carouselItems: res.data
		// 	});
		// });
		// axios.get(`` /* all the live things */).then(res => {
		// 	// this.setState({
		// 	// 	liveItems: res.data
		// 	// })
		// });
	}
	
	render() {

		const carouselItems2 = [{
			link: 'http://www.animalspot.net/wp-content/uploads/2012/05/Greater-White-Fronted-Goose-Images.jpg',
			title: 'goose1'
		},
		{
			link: 'https://myronathon.files.wordpress.com/2012/04/canada-goose-solo-flying.png',
			title: 'goose2'
		},
		{
			link : 'https://pmdvod.nationalgeographic.com/NG_Video_DEV/822/1022/goose_canada_480x360.jpg',
			title: 'goose3'
		},
		{
			link : 'http://4.bp.blogspot.com/-SM7-1JmoFWM/UgFNona2Z4I/AAAAAAAAAYg/1MRseI-VEHM/s1600/goose.jpg',
			title: 'goose4'
		}]

		return (
			<HomeOneDiv>
				<Carousel carouselItems={carouselItems2}/>
				<br/>
				<br/>
				<HomeLiveContainer>
					{carouselItems2.map((goose) => (
						<img key={`goose-${goose.title}`}style={{height: '10rem', width: '15rem', margin: '2rem'}} alt={goose.title} src={goose.link}/>
					)) }
				</HomeLiveContainer>
			</HomeOneDiv>
		);
	}
}