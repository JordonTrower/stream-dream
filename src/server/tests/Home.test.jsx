import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../../client/Components/home/Home';

Enzyme.configure({
	adapter: new Adapter()
});

it('should render Home component', () => {
	const homeComponent = shallow(<Home />);
	const tree = toJson(homeComponent);
	expect(tree).toMatchSnapshot();
})

// test('getCarouselVideos updates state with an array of carousel items', () => {

// })