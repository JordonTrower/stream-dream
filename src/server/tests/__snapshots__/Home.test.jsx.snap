// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <styled.div>
        <div
          className="sc-bZQynM iIFhze"
        >
          <Carousel
            carouselItems={Array []}
          >
            <styled.div>
              <div
                className="sc-htpNat dJnGbC"
              >
                <Styled(Slider)
                  autoplay={true}
                  autoplaySpeed={3000}
                  centerMode={true}
                  dots={true}
                  infinite={true}
                  pauseOnHover={true}
                  slidesToScroll={1}
                  slidesToShow={1}
                >
                  <Slider
                    autoplay={true}
                    autoplaySpeed={3000}
                    centerMode={true}
                    className="sc-EHOje hsamxs"
                    dots={true}
                    infinite={true}
                    pauseOnHover={true}
                    slidesToScroll={1}
                    slidesToShow={1}
                  >
                    <InnerSlider
                      accessibility={true}
                      adaptiveHeight={false}
                      afterChange={null}
                      appendDots={[Function]}
                      arrows={true}
                      autoplay={true}
                      autoplaySpeed={3000}
                      beforeChange={null}
                      centerMode={true}
                      centerPadding="50px"
                      className="sc-EHOje hsamxs"
                      cssEase="ease"
                      customPaging={[Function]}
                      dots={true}
                      dotsClass="slick-dots"
                      draggable={true}
                      easing="linear"
                      edgeFriction={0.35}
                      fade={false}
                      focusOnSelect={false}
                      infinite={true}
                      initialSlide={0}
                      lazyLoad={null}
                      nextArrow={null}
                      onEdge={null}
                      onInit={null}
                      onLazyLoadError={null}
                      onReInit={null}
                      pauseOnDotsHover={false}
                      pauseOnFocus={false}
                      pauseOnHover={true}
                      prevArrow={null}
                      responsive={null}
                      rows={1}
                      rtl={false}
                      slide="div"
                      slidesPerRow={1}
                      slidesToScroll={1}
                      slidesToShow={1}
                      speed={500}
                      swipe={true}
                      swipeEvent={null}
                      swipeToSlide={false}
                      touchMove={true}
                      touchThreshold={5}
                      unslick={true}
                      useCSS={true}
                      useTransform={true}
                      variableWidth={false}
                      vertical={false}
                      waitForAnimate={true}
                    >
                      <div
                        className="slick-slider sc-EHOje hsamxs slick-initialized"
                      >
                        <div
                          className="slick-list"
                        >
                          <Track
                            centerMode={true}
                            centerPadding="50px"
                            cssEase="ease"
                            currentSlide={0}
                            fade={false}
                            focusOnSelect={null}
                            infinite={true}
                            lazyLoad={null}
                            lazyLoadedList={Array []}
                            listHeight={0}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                            rtl={false}
                            slideCount={0}
                            slideHeight={0}
                            slideWidth={-100}
                            slidesToScroll={1}
                            slidesToShow={1}
                            speed={500}
                            trackStyle={
                              Object {
                                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                                "WebkitTransition": "",
                                "msTransform": "translateX(0px)",
                                "opacity": 1,
                                "transform": "translate3d(0px, 0px, 0px)",
                                "transition": "",
                              }
                            }
                            unslick={true}
                            variableWidth={false}
                            vertical={false}
                          >
                            <div
                              className="slick-track"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseOver={[Function]}
                              style={
                                Object {
                                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                                  "WebkitTransition": "",
                                  "msTransform": "translateX(0px)",
                                  "opacity": 1,
                                  "transform": "translate3d(0px, 0px, 0px)",
                                  "transition": "",
                                }
                              }
                            />
                          </Track>
                        </div>
                      </div>
                    </InnerSlider>
                  </Slider>
                </Styled(Slider)>
              </div>
            </styled.div>
          </Carousel>
          <br />
          <br />
          <styled.div>
            <div
              className="sc-gzVnrw hCBRPB"
            >
              <styled.div>
                <div
                  className="sc-bwzfXH cgaatZ"
                />
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Home>
  </Router>
</BrowserRouter>
`;
